RELEASE_CHECKLIST.md

```diff
@@ -0,0 +1,153 @@
+
+### Start a release PR
+
+1. Create a new branch "#.#.#" where "#.#.#" is this release's version (release) or "#.#.#-rc.#" (release candidate)
+1. Push up a commit with the `Cargo.toml`, `Cargo.lock` and `CHANGELOG.md` changes. The commit message can just be "#.#.#" (release) or "#.#.#-rc.#" (release candidate)
+1. Request review from the Apollo GraphQL tooling team.
+
+### Review
+
+Most review comments will be about the changelog. Once the PR is finalized and approved:
+
+1. If you made changes, squash or fixup all changes into a single commit.
+1. Run `git push` and wait for CI to pass.
+
+### Tag and build release
+
+This part of the release process is handled by GitHub Actions, and our binaries are distributed as GitHub Releases. When you push a version tag, it kicks off an action that creates a new GitHub release for that tag, builds release binaries and attaches them to the release.
+
+1. Once ready to merge, tag the commit by running either `git tag -a v#.#.# -m #.#.#` (release), or `git tag -a v#.#.#-rc.# -m #.#.#` (release candidate)
+1. Run `git push --tags`.
+1. Wait for CI to pass.
+
+### Edit the release
+
+After CI builds the release binaries and they appear on the [releases page](https://github.com/apollographql/apollo-cli/releases), click `Edit` and update release notes.
+
+#### For Full Releases
+
+1. Paste the current release notes from `CHANGELOG.md` into the release body.
+1. Update the *title* of the release (not the tag itself) to include the emoji for the current release
+1. Be sure to add any missing link definitions to the release.
+
+#### For Release Candidates
+
+1. CI should already mark the release as a `pre-release`. Double check that it's listed as a pre-release on the release's Edit page.
+1. If this is a new rc (rc.0), paste testing instructions into the release notes.
+1. If this is a rc.1 or later, the old release candidate testing instructions should be moved to the latest release candidate testing instructions, and replaced with the following message:
+
+   ```markdown
+   This beta release is now out of date. If you previously installed this release, you should reinstall and see what's changed in the latest [release](https://github.com/apollographql/apollo-cli/releases).
+   ```
+
+   The new release candidate should then include updated testing instructions with a small changelog at the top to get folks who installed the old release candidate up to speed.
+
+## Publish
+
+1. Hit the big green Merge button on the release PR.
+1. `git checkout master` and `git pull --rebase origin master`
+
+### Publish to crates.io (full release only)
+
+**IMPORTANT: This step is the hardest to fix if you mess it up. Do not run this step for Release Candidates**.
+
+We don't publish release candidates to crates.io because they don't (as of this writing) have a concept of a "beta" version.
+
+1. Run `cargo test`
```

Comment:
Since we don't have cargo/npm yet should we hold off on this? Or go ahead and add it in since we know we are doing this?

Reply:
I figured we'd add it since we know we're going to do it, and you'd also want to run these first couple of commands even if you weren't publishing to crates.io